plugins/Login/Login.php

```diff
@@ -219,11 +219,13 @@ protected static function isModuleIsAPI()
     private function getUsernameUsedInPasswordLogin()
     {
         $login = StaticContainer::get(\Piwik\Auth::class)->getLogin();
-        if (empty($login)
-            || $login == 'anonymous'
-        ) {
+        if (empty($login) || $login == 'anonymous') {
             $login = Common::getRequestVar('form_login', false);
         }
+        $action = Common::getRequestVar('action', false);
+        if ((empty($login) || $login == 'anonymous') && $action == 'logme') {
+            $login = Common::getRequestVar('login', false);
+        }
```

Comment:
Doing `empty($login) || $login == 'anonymous'` seems a bit redundant. Would it be worth it to simplify that?

Also, do you think it makes sense to move this function to another class, for example, BruteForceDetection so it can be tested?

Reply:
Will simplify the code. Moving it actually wouldn't make it much better I guess.

Reply:
I was thinking moving might help so we could cover it in tests. Do you think testing it would be worth the trouble?